classes/script_metadata.php

```diff
@@ -316,4 +316,15 @@ public static function get_sampling_period(): float {
         return round($insensiblerange ? $period : self::SAMPLING_PERIOD_DEFAUILT, 3);
     }
 
+    /**
+     * Returns the sampling double rate.
+     *
+     * @return int
+     * @throws \dml_exception
+     */
+    public static function get_sampling_doublerate(): int {
+        $rate = (int)get_config('tool_excimer', 'doublerate');
+        $insensiblerange = $rate >= 0;
+        return $insensiblerange ? $rate : 1024;
```

Comment:
Was a bit hard to read, please consider the following

```suggestion
        if ($rate <= 0) {
            return 1024;
        }
        return $rate;
```